Src/VimCore/UndoRedoOperations.fs

```diff
@@ -363,16 +367,28 @@ and UndoRedoOperations
                 | UndoRedoData.Linked (count, false) :: tail -> 
                     Contract.Assert(count > 0)
                     hasData <- true
-                    _undoStack <- UndoRedoData.Linked (count, true) :: tail
+
+                    // If the linked transaction ended with an insert, then append an empty
+                    // transaction to the previous undo transaction to break the undo sequence.
+                    let extra =
+                        if endsWithInsert then
+                            use undoTransaction = textUndoHistory.CreateTransaction "Break undo sequence"
+                            undoTransaction.Complete()
+                            1
+                        else
+                            0
+
+                    _undoStack <- UndoRedoData.Linked (count + extra, true) :: tail

```

Comment:
After thinking more about this I do agree this is the best approach. There really isn't a better option here. 

I considered ways we could detect this case and add the transaction only when needed. But given the VS implementation of `ITextUndoHistory` is largely unimplemented (never exposes a transaction directly) there really isn't much of an option. 

This is just kind of how it is with `ITextUndoHistory`. The API is actually really nice at a glance but the VS implementation is the one that matters and it effectively removes most of the API surface.